src/bindings/python/src/openvino/runtime/ie_api.py

```diff
@@ -49,8 +46,9 @@ def map_tensor_names_to_types(input: Output) -> dict:
         return {n: input.get_element_type() for n in input.get_names()}
 
     input_types: dict = {}
-    for input in obj.inputs:
-        input_types = {**input_types, **map_tensor_names_to_types(input)}
+    for idx, input in enumerate(obj.inputs):
+        input_types.update(map_tensor_names_to_types(input))
+        input_types[idx] = input.get_element_type()
```

Comment:
How this improve above solution? Some edge case or performance gain?

Reply:
firstly, in some corner cases, tensors can not have names, then map_tensor_names_to_types(input) return empty dict for this scenario and we will be not able to normalize input for this case
secondly, if provided indexes for all inputs as keys for feed dict for inference, previous behavior for getting type by order give mismatch if we have several names for the same input tensor